### ISSUE TITLE
element does not have a [lang] attribute

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#9634]

**Comment by veronicamar** [comment]
From PageSpeed:
If a page doesn't specify a lang attribute, a screen reader assumes that the page is in the default language that the user chose when setting up the screen reader. If the page isn't actually in the default language, then the screen reader might not announce the page's text correctly.

<img width="905" alt="Screenshot 2024-07-27 at 2 56 07 pm" src="https://github.com/user-attachments/assets/4d022df2-2a6f-4376-abd6-90c725c67da8">

Add a lang attribute to the html element (e.g. <html lang="en">) whose value represents the primary language of document.

Make sure you identify a language in the opening <html> element and spell the attribute correctly. For example, if the primary language of a document is English, you could specify the language as follows:

< html lang="en' > 
  < !--document head and body-- >
< /html >

This can be specified for dialects with codes like 'en-US' for American English- if the language changes within the document it can be specified:

< p > Text in one language < span lang="es" > text in another language < /span > < /p >

**Comment by srivastavaarpit977** [comment]
Hey @cdrini @veronicamar I would like to work on this issue..plz assign this issue to me

**Comment by Jash2606** [comment]
Hey @cdrini , I would like to work on this issue. please could you assign me this issue?

**Comment by Jash2606** [comment]
Hey @scottbarnes @cdrini , I have updated the lang attribute as mentioned in issue.
Please review the changes and let me know if any further updates are needed.
Thank you!

**Comment by scottbarnes** [comment]
I wonder if the root cause of this may be that the PageSpeed insights bot isn't sending the `Accept-Language` header, and that `lang` attribute is otherwise properly being set here: https://github.com/internetarchive/openlibrary/blob/04fb777459d99c4e1010f46af51fc37d98df1e41/openlibrary/templates/site/head.html#L4

Consider the following (I removed some blank spaces from the output):

`curl` with no headers set:
```
curl -s https://openlibrary.org | head -n 5 
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="">
<head>
```
`curl` with `Accept-Language` set for Spanish.
```
curl -s -H "Accept-Language: es" https://openlibrary.org | head -n 5
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es">
<head>
```
I noticed when I visit Open Library using Firefox, the browser is sending `Accept-Language: en-US,en;q=0.5`, and the page displays in English, and in the page's source, I see:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
```
There may be some benefit in having this default to English when no language is set, if only so bots get a `lang` value that matches what the page will be, which is English. Thoughts?

**Comment by Jash2606** [comment]
> I wonder if the root cause of this may be that the PageSpeed insights bot isn't sending the `Accept-Language` header, and that `lang` attribute is otherwise properly being set here:
> 
> 
### Proposed Solution
There is a benefit in having the `lang` attribute default to English when no language is set. This ensures that bots, like PageSpeed Insights, receive a `lang` value that matches the page's content, which is English.

### Implementation Suggestion
Modify the template in `openlibrary/openlibrary/templates/site/head.html` to default the `lang` attribute to English if no language is detected. This can be done using a conditional check in the template language.

```html
<html xmlns="http://www.w3.org/1999/xhtml" lang="${get_lang() or 'en'}">
```

This ensures that the `lang` attribute is always set to a valid value, improving accessibility and providing correct information to bots and screen readers.

@scottbarnes  Should I go for it ?

**Comment by scottbarnes** [comment]
@Jash2606, that seems a reasonable solution, though perhaps others will disagree. But since it would take minimal time, please feel free to update your existing PR to make the change you've proposed.

**Comment by Jash2606** [comment]
> @Jash2606, that seems a reasonable solution, though perhaps others will disagree. But since it would take minimal time, please feel free to update your existing PR to make the change you've proposed.



Hi @scottbarnes ,

I have updated the existing PR with the proposed changes . Could you please discuss this with the team at your earliest convenience and let me know if any further changes are needed?
